--- Makefile.am	2019-01-05 00:50:52.383807000 +0000
+++ Makefile.am.new	2019-01-05 00:51:18.923807000 +0000
@@ -168,6 +168,12 @@
 CORE_CFLAGS += -DSWITCH_HAVE_FREETYPE $(LIBFREETYPE_CFLAGS)
 endif
 
+# DH: GRPC
+if HAVE_GRPC
+CORE_CFLAGS += -DSWITCH_HAVE_GRPC $(LIBGRPC_CFLAGS)
+GRPC_LIBS += -lgrpc++_reflection -lprotobuf 
+endif
+
 ##
 ## libfreeswitch
 ##
@@ -175,6 +181,9 @@
 if ENABLE_LIBYUV
 noinst_LTLIBRARIES += libfreeswitch_libyuv.la
 endif
+if HAVE_GRPC
+noinst_LTLIBRARIES += libfreeswitch_libgoogleapis.la
+endif
 libfreeswitch_spandsp_la_SOURCES = libs/spandsp/src/plc.c libs/spandsp/src/alloc.c libs/spandsp/src/bit_operations.c
 libfreeswitch_spandsp_la_CFLAGS  = -Ilibs/spandsp/src $(CORE_CFLAGS) $(AM_CFLAGS)
 CORE_LIBS+=libfreeswitch_spandsp.la
@@ -225,14 +234,166 @@
     libs/libyuv/source/scale_neon.cc        \
     libs/libyuv/source/scale_win.cc
 
+
 libfreeswitch_libyuv_la_CPPFLAGS = -O2 -fomit-frame-pointer -Ilibs/libyuv/include
 CORE_LIBS+=libfreeswitch_libyuv.la
 endif
 
+if HAVE_GRPC
+GOOGLEAPIS_GENS_PATH = libs/googleapis/gens
+GOOGLEAPIS_LOGGING_CCS = $(shell find $(GOOGLEAPIS_GENS_PATH)/google/logging \
+	-name '*.pb.cc')
+GOOGLEAPIS_API_CCS = $(shell find $(GOOGLEAPIS_GENS_PATH)/google/api \
+	-name '*.pb.cc')
+GOOGLEAPIS_RPC_CCS = $(shell find $(GOOGLEAPIS_GENS_PATH)/google/rpc \
+	-name '*.pb.cc')
+GOOGLEAPIS_SPEECH_CCS = $(shell find \
+	$(GOOGLEAPIS_GENS_PATH)/google/cloud/speech -name '*.pb.cc')
+GOOGLEAPIS_LONGRUNNING_CCS = $(shell find \
+	$(GOOGLEAPIS_GENS_PATH)/google/longrunning -name '*.pb.cc')
+GOOGLEAPIS_CCS = $(GOOGLEAPIS_API_CCS) $(GOOGLEAPIS_RPC_CCS) \
+	$(GOOGLEAPIS_LONGRUNNING_CCS) $(GOOGLEAPIS_SPEECH_CCS)
+
+nodist_libfreeswitch_libgoogleapis_la_SOURCES = \
+	libs/googleapis/gens/google/api/monitoring.grpc.pb.cc \
+	libs/googleapis/gens/google/api/annotations.grpc.pb.cc \
+	libs/googleapis/gens/google/api/http.pb.cc \
+	libs/googleapis/gens/google/api/quota.pb.cc \
+	libs/googleapis/gens/google/api/quota.grpc.pb.cc \
+	libs/googleapis/gens/google/api/backend.grpc.pb.cc \
+	libs/googleapis/gens/google/api/service.grpc.pb.cc \
+	libs/googleapis/gens/google/api/monitored_resource.pb.cc \
+	libs/googleapis/gens/google/api/consumer.pb.cc \
+	libs/googleapis/gens/google/api/annotations.pb.cc \
+	libs/googleapis/gens/google/api/metric.pb.cc \
+	libs/googleapis/gens/google/api/logging.pb.cc \
+	libs/googleapis/gens/google/api/experimental/authorization_config.pb.cc \
+	libs/googleapis/gens/google/api/experimental/authorization_config.grpc.pb.cc \
+	libs/googleapis/gens/google/api/experimental/experimental.grpc.pb.cc \
+	libs/googleapis/gens/google/api/experimental/experimental.pb.cc \
+	libs/googleapis/gens/google/api/auth.grpc.pb.cc \
+	libs/googleapis/gens/google/api/distribution.grpc.pb.cc \
+	libs/googleapis/gens/google/api/label.grpc.pb.cc \
+	libs/googleapis/gens/google/api/launch_stage.grpc.pb.cc \
+	libs/googleapis/gens/google/api/launch_stage.pb.cc \
+	libs/googleapis/gens/google/api/httpbody.grpc.pb.cc \
+	libs/googleapis/gens/google/api/config_change.grpc.pb.cc \
+	libs/googleapis/gens/google/api/logging.grpc.pb.cc \
+	libs/googleapis/gens/google/api/context.pb.cc \
+	libs/googleapis/gens/google/api/system_parameter.pb.cc \
+	libs/googleapis/gens/google/api/distribution.pb.cc \
+	libs/googleapis/gens/google/api/control.pb.cc \
+	libs/googleapis/gens/google/api/consumer.grpc.pb.cc \
+	libs/googleapis/gens/google/api/label.pb.cc \
+	libs/googleapis/gens/google/api/documentation.pb.cc \
+	libs/googleapis/gens/google/api/log.pb.cc \
+	libs/googleapis/gens/google/api/usage.grpc.pb.cc \
+	libs/googleapis/gens/google/api/backend.pb.cc \
+	libs/googleapis/gens/google/api/control.grpc.pb.cc \
+	libs/googleapis/gens/google/api/log.grpc.pb.cc \
+	libs/googleapis/gens/google/api/source_info.grpc.pb.cc \
+	libs/googleapis/gens/google/api/billing.pb.cc \
+	libs/googleapis/gens/google/api/auth.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/service_controller.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/check_error.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/check_error.grpc.pb.cc \ 
+	libs/googleapis/gens/google/api/servicecontrol/v1/distribution.grpc.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/quota_controller.grpc.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/metric_value.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/distribution.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/log_entry.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/service_controller.grpc.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/metric_value.grpc.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/log_entry.grpc.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/operation.grpc.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/quota_controller.pb.cc \
+	libs/googleapis/gens/google/api/servicecontrol/v1/operation.pb.cc \
+	libs/googleapis/gens/google/api/metric.grpc.pb.cc \
+	libs/googleapis/gens/google/api/monitored_resource.grpc.pb.cc \
+	libs/googleapis/gens/google/api/http.grpc.pb.cc \
+	libs/googleapis/gens/google/api/httpbody.pb.cc \
+	libs/googleapis/gens/google/api/endpoint.pb.cc \
+	libs/googleapis/gens/google/api/documentation.grpc.pb.cc \
+	libs/googleapis/gens/google/api/system_parameter.grpc.pb.cc \
+	libs/googleapis/gens/google/api/endpoint.grpc.pb.cc \
+	libs/googleapis/gens/google/api/service.pb.cc \
+	libs/googleapis/gens/google/api/source_info.pb.cc \
+	libs/googleapis/gens/google/api/servicemanagement/v1/resources.grpc.pb.cc \
+	libs/googleapis/gens/google/api/servicemanagement/v1/servicemanager.pb.cc \
+	libs/googleapis/gens/google/api/servicemanagement/v1/resources.pb.cc \
+	libs/googleapis/gens/google/api/servicemanagement/v1/servicemanager.grpc.pb.cc \
+	libs/googleapis/gens/google/api/billing.grpc.pb.cc \
+	libs/googleapis/gens/google/api/usage.pb.cc \
+	libs/googleapis/gens/google/api/config_change.pb.cc \
+	libs/googleapis/gens/google/api/context.grpc.pb.cc \
+	libs/googleapis/gens/google/api/monitoring.pb.cc \
+	libs/googleapis/gens/google/rpc/error_details.grpc.pb.cc \
+	libs/googleapis/gens/google/rpc/code.pb.cc \
+	libs/googleapis/gens/google/rpc/status.pb.cc \
+	libs/googleapis/gens/google/rpc/status.grpc.pb.cc \
+	libs/googleapis/gens/google/rpc/error_details.pb.cc \
+	libs/googleapis/gens/google/rpc/code.grpc.pb.cc \
+	libs/googleapis/gens/google/longrunning/operations.grpc.pb.cc \
+	libs/googleapis/gens/google/longrunning/operations.pb.cc \
+	libs/googleapis/gens/google/cloud/speech/v1/cloud_speech.pb.cc \
+	libs/googleapis/gens/google/cloud/speech/v1/cloud_speech.grpc.pb.cc \
+	libs/googleapis/gens/google/logging/type/http_request.grpc.pb.cc  \
+	libs/googleapis/gens/google/logging/type/log_severity.grpc.pb.cc \
+	libs/googleapis/gens/google/logging/type/log_severity.pb.cc \
+	libs/googleapis/gens/google/logging/type/http_request.pb.cc \
+	libs/googleapis/gens/google/logging/v2/logging.pb.cc  \
+	libs/googleapis/gens/google/logging/v2/logging_metrics.pb.cc \
+	libs/googleapis/gens/google/logging/v2/logging.grpc.pb.cc \
+	libs/googleapis/gens/google/logging/v2/log_entry.pb.cc \
+	libs/googleapis/gens/google/logging/v2/logging_config.grpc.pb.cc \
+	libs/googleapis/gens/google/logging/v2/logging_config.pb.cc \
+	libs/googleapis/gens/google/logging/v2/log_entry.grpc.pb.cc \
+	libs/googleapis/gens/google/logging/v2/logging_metrics.grpc.pb.cc \
+	libs/googleapis/gens/google/type/date.grpc.pb.cc \
+	libs/googleapis/gens/google/type/timeofday.pb.cc \
+	libs/googleapis/gens/google/type/latlng.grpc.pb.cc \
+	libs/googleapis/gens/google/type/money.pb.cc \
+	libs/googleapis/gens/google/type/date.pb.cc \
+	libs/googleapis/gens/google/type/postal_address.grpc.pb.cc \
+	libs/googleapis/gens/google/type/dayofweek.grpc.pb.cc \
+	libs/googleapis/gens/google/type/dayofweek.pb.cc \
+	libs/googleapis/gens/google/type/timeofday.grpc.pb.cc \
+	libs/googleapis/gens/google/type/color.pb.cc \
+	libs/googleapis/gens/google/type/postal_address.pb.cc \
+	libs/googleapis/gens/google/type/latlng.pb.cc \
+	libs/googleapis/gens/google/type/color.grpc.pb.cc \
+	libs/googleapis/gens/google/type/money.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/agent.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/agent.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/audio_config.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/audio_config.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/context.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/context.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/document.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/document.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/entity_type.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/entity_type.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/intent.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/intent.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/knowledge_base.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/knowledge_base.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/session_entity_type.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/session_entity_type.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/session.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/session.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/webhook.grpc.pb.cc \
+	libs/googleapis/gens/google/cloud/dialogflow/v2beta1/webhook.pb.cc
+
+libfreeswitch_libgoogleapis_la_CPPFLAGS = -I/usr/local/include -I$(GOOGLEAPIS_GENS_PATH) -std=c++11  -pthread
+
+CORE_LIBS+=libfreeswitch_libgoogleapis.la
+endif
+
+
 lib_LTLIBRARIES	         = libfreeswitch.la
-libfreeswitch_la_CFLAGS  = $(CORE_CFLAGS) $(SQLITE_CFLAGS) $(FREETYPE_CFLAGS) $(CURL_CFLAGS) $(PCRE_CFLAGS) $(SPEEX_CFLAGS) $(LIBEDIT_CFLAGS) $(openssl_CFLAGS)  $(AM_CFLAGS)
+libfreeswitch_la_CFLAGS  = $(CORE_CFLAGS) $(GRPC_CFLAGS) $(SQLITE_CFLAGS) $(FREETYPE_CFLAGS) $(CURL_CFLAGS) $(PCRE_CFLAGS) $(SPEEX_CFLAGS) $(LIBEDIT_CFLAGS) $(openssl_CFLAGS) $(AM_CFLAGS)
 libfreeswitch_la_LDFLAGS = -version-info 1:0:0 $(AM_LDFLAGS) $(PLATFORM_CORE_LDFLAGS) -no-undefined
-libfreeswitch_la_LIBADD  = $(CORE_LIBS) $(APR_LIBS) $(SQLITE_LIBS) $(FREETYPE_LIBS) $(CURL_LIBS) $(PCRE_LIBS) $(SPEEX_LIBS) $(LIBEDIT_LIBS) $(openssl_LIBS) $(PLATFORM_CORE_LIBS)
+libfreeswitch_la_LIBADD  = $(CORE_LIBS) $(APR_LIBS) $(GRPC_LIBS) $(SQLITE_LIBS) $(FREETYPE_LIBS) $(CURL_LIBS) $(PCRE_LIBS) $(SPEEX_LIBS) $(LIBEDIT_LIBS) $(openssl_LIBS) $(GRPC_LIBS) $(PLATFORM_CORE_LIBS)
 libfreeswitch_la_DEPENDENCIES = $(BUILT_SOURCES)
 
 if HAVE_PNG
@@ -471,7 +632,6 @@
 freeswitch_LDADD += $(ODBC_LIB_FLAGS)
 endif
 
-
 ##
 ## Scripts
 ##
