--- configure.ac	2019-02-06 17:35:27.476000000 +0000
+++ configure.ac.new	2019-02-06 17:35:19.032000000 +0000
@@ -13,7 +13,8 @@
 AC_CONFIG_FILES([src/include/switch_version.h.in:src/include/switch_version.h.template])

 AC_CONFIG_AUX_DIR(build/config)
-AM_INIT_AUTOMAKE
+#AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([subdir-objects])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_CONFIG_SRCDIR([src/switch.c])
 AC_CONFIG_HEADER([src/include/switch_private.h])
@@ -1475,6 +1476,34 @@
       AC_MSG_ERROR([You need to either install libedit-dev (>= 2.11) or configure with --disable-core-libedit-support])
       ])])])

+dnl DH: Added for including libwebsockets
+AC_ARG_WITH(lws,
+  [AS_HELP_STRING([--with-lws],
+    [enable support for libwebsockets])],
+  [with_lws="$withval"],
+  [with_lws="no"])
+if test "$with_lws" = "yes"; then
+  PKG_CHECK_MODULES([LWS], [libwebsockets], [
+    AM_CONDITIONAL([HAVE_LWS],[true])], [
+    AC_MSG_RESULT([no]); AM_CONDITIONAL([HAVE_LWS],[false])])
+else
+  AM_CONDITIONAL([HAVE_LWS],[false])
+fi
+
+dnl DH: Added for including google protobuf libs
+AC_ARG_WITH(grpc,
+  [AS_HELP_STRING([--with-grpc],
+    [enable support for google rpc (libgrpc++ and libgrpc)])],
+  [with_grpc="$withval"],
+  [with_grpc="no"])
+if test "$with_grpc" = "yes"; then
+  PKG_CHECK_MODULES([GRPC], [grpc++ grpc], [
+    AM_CONDITIONAL([HAVE_GRPC],[true])], [
+    AC_MSG_RESULT([no]); AM_CONDITIONAL([HAVE_GRPC],[false])])
+else
+  AM_CONDITIONAL([HAVE_GRPC],[false])
+fi
+
 dnl ---------------------------------------------------------------------------
 dnl - OpenLDAP SDK
 dnl ---------------------------------------------------------------------------
@@ -1767,6 +1796,9 @@
 		src/mod/applications/mod_curl/Makefile
 		src/mod/applications/mod_cv/Makefile
 		src/mod/applications/mod_db/Makefile
+		src/mod/applications/mod_google_tts/Makefile
+		src/mod/applications/mod_google_transcribe/Makefile
+		src/mod/applications/mod_dialogflow/Makefile
 		src/mod/applications/mod_directory/Makefile
 		src/mod/applications/mod_distributor/Makefile
 		src/mod/applications/mod_dptools/Makefile
