--- configure.ac	2019-01-06 18:14:19.221847000 -0500
+++ configure.ac.new	2019-01-06 20:47:23.381847000 -0500
@@ -1490,6 +1490,20 @@
   AM_CONDITIONAL([HAVE_GRPC],[false])
 fi
 
+dnl DH: Added for including libwebsockets
+AC_ARG_WITH(lws,
+  [AS_HELP_STRING([--with-lws],
+    [enable support for libwebsockets])],
+  [with_lws="$withval"],
+  [with_lws="no"])
+if test "$with_lws" = "yes"; then
+  PKG_CHECK_MODULES([LWS], [libwebsockets], [
+    AM_CONDITIONAL([HAVE_LWS],[true])], [
+    AC_MSG_RESULT([no]); AM_CONDITIONAL([HAVE_LWS],[false])])
+else
+  AM_CONDITIONAL([HAVE_LWS],[false])
+fi
+
 dnl ---------------------------------------------------------------------------
 dnl - OpenLDAP SDK
 dnl ---------------------------------------------------------------------------
